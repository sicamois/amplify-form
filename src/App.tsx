import React from 'react';
import './App.css';

function App() {
  return (
    <div className="App">
      <body>
        <h1 className='text-xl text-center font-black'>Test AmplifyForm</h1>
      </body>
    </div>
  );
}

export default App;
