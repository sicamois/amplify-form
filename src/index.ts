// @ts-ignore
import styleInject from 'style-inject';
// @ts-ignore
import css from './styles/tailwind.css';
styleInject(css);
export { default } from './components/AmplifyForm/AmplifyForm';
export * from './types';
