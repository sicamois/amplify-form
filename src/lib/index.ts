import AmplifyForm from "./components/AmplifyForm";

export { AmplifyForm } 