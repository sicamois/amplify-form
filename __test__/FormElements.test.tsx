import { Form, Formik, FormikProps } from 'formik';
import { FC } from 'react';
import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import {
  CheckboxField,
  NumberField,
  TextAreaField,
  TextField,
} from '../src/components/FormElements';
import { FieldProps, FormTheme } from '../src/types';

const TestForm: FC<{ initialValues: Record<string, any> }> = ({
  initialValues,
  children,
}) => {
  return (
    <Formik
      enableReinitialize
      initialValues={initialValues}
      onSubmit={() => {}}>
      {() => <Form>{children}</Form>}
    </Formik>
  );
};

describe('TextField', () => {
  let spy: jest.SpyInstance;
  let spyError: jest.SpyInstance;
  let spyWarn: jest.SpyInstance;
  beforeAll(() => {
    spy = jest.spyOn(console, 'log').mockImplementation(() => {});
    spyError = jest.spyOn(console, 'error').mockImplementation(() => {});
    spyWarn = jest.spyOn(console, 'warn').mockImplementation(() => {});
  });
  afterAll(() => {
    spy.mockRestore();
    spyError.mockRestore();
    spyWarn.mockRestore();
  });

  it('renders correctly', () => {
    const props: FieldProps = {
      name: 'title',
    };
    const tree = create(
      <TestForm initialValues={{ title: '' }}>
        <TextField {...props} />
      </TestForm>
    ).toJSON();
    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('sets field size correctly when prop is passed', () => {
    const name = 'title';
    const props: FieldProps = {
      name,
      fieldSize: 'md',
    };
    render(
      <TestForm initialValues={{ title: '' }}>
        <TextField {...props} />
      </TestForm>
    );
    const inputElement = screen.getByRole('textbox');
    expect(inputElement.classList).toContain('w-36');
    expect(inputElement.classList).not.toContain('w-full');
  });

  it('set field size to default when prop is set incorrectly', () => {
    const name = 'title';
    const props: FieldProps = {
      name,
      fieldSize: 'xxx',
    };
    render(
      <TestForm initialValues={{ title: '' }}>
        <TextField {...props} />
      </TestForm>
    );
    const inputElement = screen.getByRole('textbox');
    expect(inputElement.classList).toContain('w-full');
  });

  it('sets basic theme correctly when prop is passed', () => {
    const name = 'title';
    const theme: FormTheme = { color: 'lime' };
    const props: FieldProps = {
      name,
      theme,
    };
    render(
      <TestForm initialValues={{ title: '' }}>
        <TextField {...props} />
      </TestForm>
    );
    const inputElement = screen.getByRole('textbox');
    expect(inputElement.classList).toContain('focus:border-lime-600');
    expect(inputElement.classList).not.toContain('focus:border-red-900');
  });
});

describe('TextAreaField', () => {
  let spy: jest.SpyInstance;
  let spyError: jest.SpyInstance;
  let spyWarn: jest.SpyInstance;
  beforeAll(() => {
    spy = jest.spyOn(console, 'log').mockImplementation(() => {});
    spyError = jest.spyOn(console, 'error').mockImplementation(() => {});
    spyWarn = jest.spyOn(console, 'warn').mockImplementation(() => {});
  });
  afterAll(() => {
    spy.mockRestore();
    spyError.mockRestore();
    spyWarn.mockRestore();
  });

  it('renders correctly', () => {
    const props: FieldProps = {
      name: 'title',
    };
    const tree = create(
      <TestForm initialValues={{ title: '' }}>
        <TextAreaField {...props} />
      </TestForm>
    ).toJSON();
    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('sets field size correctly when prop is passed', () => {
    const name = 'title';
    const props: FieldProps = {
      name,
      fieldSize: 'md',
    };
    render(
      <TestForm initialValues={{ title: '' }}>
        <TextAreaField {...props} />
      </TestForm>
    );
    const inputElement = screen.getByRole('textbox');
    expect(inputElement.classList).toContain('w-36');
    expect(inputElement.classList).not.toContain('w-100');
  });

  it('set field size to default when prop is set incorrectly', () => {
    const name = 'title';
    const props: FieldProps = {
      name,
      fieldSize: 'xxx',
    };
    render(
      <TestForm initialValues={{ title: '' }}>
        <TextAreaField {...props} />
      </TestForm>
    );
    const inputElement = screen.getByRole('textbox');
    expect(inputElement.classList).toContain('w-100');
  });

  it('sets basic theme correctly when prop is passed', () => {
    const name = 'title';
    const theme: FormTheme = { color: 'lime' };
    const props: FieldProps = {
      name,
      theme,
    };
    render(
      <TestForm initialValues={{ title: '' }}>
        <TextAreaField {...props} />
      </TestForm>
    );
    const inputElement = screen.getByRole('textbox');
    expect(inputElement.classList).toContain('focus:border-lime-600');
    expect(inputElement.classList).not.toContain('focus:border-red-900');
  });
});

describe('NumberField', () => {
  let spy: jest.SpyInstance;
  let spyError: jest.SpyInstance;
  let spyWarn: jest.SpyInstance;
  beforeAll(() => {
    spy = jest.spyOn(console, 'log').mockImplementation(() => {});
    spyError = jest.spyOn(console, 'error').mockImplementation(() => {});
    spyWarn = jest.spyOn(console, 'warn').mockImplementation(() => {});
  });
  afterAll(() => {
    spy.mockRestore();
    spyError.mockRestore();
    spyWarn.mockRestore();
  });

  it('renders correctly', () => {
    const props: FieldProps = {
      name: 'ref',
    };
    const tree = create(
      <TestForm initialValues={{ ref: '' }}>
        <NumberField {...props} />
      </TestForm>
    ).toJSON();
    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('sets field size correctly when prop is passed', () => {
    const name = 'ref';
    const props: FieldProps = {
      name,
      fieldSize: 'sm',
    };
    render(
      <TestForm initialValues={{ ref: '' }}>
        <NumberField {...props} />
      </TestForm>
    );
    const inputElement = screen.getByRole('spinbutton');
    expect(inputElement.classList).toContain('w-20');
    expect(inputElement.classList).not.toContain('w-36');
  });

  it('set field size to default when prop is set incorrectly', () => {
    const name = 'ref';
    const props: FieldProps = {
      name,
      fieldSize: 'xxx',
    };
    render(
      <TestForm initialValues={{ ref: '' }}>
        <NumberField {...props} />
      </TestForm>
    );
    const inputElement = screen.getByRole('spinbutton');
    expect(inputElement.classList).toContain('w-36');
  });

  it('sets basic theme correctly when prop is passed', () => {
    const name = 'ref';
    const theme: FormTheme = { color: 'lime' };
    const props: FieldProps = {
      name,
      theme,
    };
    render(
      <TestForm initialValues={{ ref: '' }}>
        <NumberField {...props} />
      </TestForm>
    );
    const inputElement = screen.getByRole('spinbutton');
    expect(inputElement.classList).toContain('focus:border-lime-600');
    expect(inputElement.classList).not.toContain('focus:border-red-900');
  });

  it('sets step correctly when prop is passed', () => {
    const name = 'ref';
    const props: FieldProps = {
      name,
      step: 0.01,
    };
    render(
      <TestForm initialValues={{ ref: '' }}>
        <NumberField {...props} />
      </TestForm>
    );
    const inputElement = screen.getByRole('spinbutton') as HTMLInputElement;
    expect(inputElement.step).toBe(props.step.toString());
  });
});

describe('CheckboxField', () => {
  let spy: jest.SpyInstance;
  let spyError: jest.SpyInstance;
  let spyWarn: jest.SpyInstance;
  beforeAll(() => {
    spy = jest.spyOn(console, 'log').mockImplementation(() => {});
    spyError = jest.spyOn(console, 'error').mockImplementation(() => {});
    spyWarn = jest.spyOn(console, 'warn').mockImplementation(() => {});
  });
  afterAll(() => {
    spy.mockRestore();
    spyError.mockRestore();
    spyWarn.mockRestore();
  });

  it('renders correctly', () => {
    const props: FieldProps = {
      name: 'public',
    };
    const tree = create(
      <TestForm initialValues={{ public: '' }}>
        <CheckboxField {...props} />
      </TestForm>
    ).toJSON();
    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('sets field size correctly when prop is passed', () => {
    const name = 'public';
    const props: FieldProps = {
      name,
      fieldSize: 'sm',
    };
    render(
      <TestForm initialValues={{ public: '' }}>
        <CheckboxField {...props} />
      </TestForm>
    );
    const divElement = screen.getByTestId(`${name}-div`);
    expect(divElement.classList).toContain('w-20');
    expect(divElement.classList).not.toContain('w-36');
  });

  it('set field size to default when prop is set incorrectly', () => {
    const name = 'public';
    const props: FieldProps = {
      name,
      fieldSize: 'xxx',
    };
    render(
      <TestForm initialValues={{ public: '' }}>
        <CheckboxField {...props} />
      </TestForm>
    );
    const divElement = screen.getByTestId(`${name}-div`);
    expect(divElement.classList).toContain('w-36');
  });

  it('sets basic theme correctly when prop is passed', () => {
    const name = 'public';
    const theme: FormTheme = { color: 'lime' };
    const props: FieldProps = {
      name,
      theme,
    };
    render(
      <TestForm initialValues={{ public: '' }}>
        <CheckboxField {...props} />
      </TestForm>
    );
    const inputElement = screen.getByRole('checkbox');
    expect(inputElement.classList).toContain('accent-lime-600');
    expect(inputElement.classList).not.toContain('accent-red-900');
  });
});
