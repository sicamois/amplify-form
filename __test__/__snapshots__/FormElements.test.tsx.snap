// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxField renders correctly 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="py-2"
  >
    <div
      className="relative"
    >
      <div
        className="relative p-3 flex justify-center w-36"
        data-testid="public-div"
      >
        <input
          checked={false}
          className="accent-red-900 bg-transparent border-gray-300 rounded border-2 w-5 h-5 focus:ring-0"
          id="public"
          name="public"
          onBlur={[Function]}
          onChange={[Function]}
          type="checkbox"
          value=""
        />
      </div>
    </div>
  </div>
</form>
`;

exports[`FieldSet renders correctly 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <fieldset
    className=" flex flex-wrap flex-row justify-start border-2 p-4 gap-3 w-full"
    name="set"
  >
    <legend
      className="text-red-900 font-black text-lg px-2"
      data-testid="set-legend"
    >
      Infos
    </legend>
    <div />
  </fieldset>
</form>
`;

exports[`FieldWithError renders correctly with label 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="py-2"
  >
    <div
      className="relative"
    >
      <div />
      <label
        className="text-gray-400 px-2 absolute left-0 -top-3.5 text-sm transition-all peer-placeholder-shown:text-base peer-placeholder-shown:text-gray-400 peer-placeholder-shown:top-2 peer-focus:-top-3.5 peer-focus:text-sm peer-focus:text-red-900  w-full"
        htmlFor="set"
      >
        Infos
      </label>
    </div>
  </div>
</form>
`;

exports[`FieldWithError renders correctly without label 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="py-2"
  >
    <div
      className="relative"
    >
      <div />
    </div>
  </div>
</form>
`;

exports[`FilesDropField renders correctly 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <fieldset
    className=" flex flex-wrap flex-row justify-start border-2 p-4 gap-3 w-full"
  >
    <legend
      className="text-red-900 font-black text-lg px-2"
    />
    <div
      className="px-2"
    >
      <div
        className="py-2"
      >
        <div
          className="relative"
        >
          <section
            className="container"
          >
            <div
              className="bg-gray-200 border-gray-600  border-2 border-dashed h-24 px-4 py-2 text-sm font-light w-[60vw]"
              onBlur={[Function]}
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <input
                autoComplete="off"
                name="gallery"
                onChange={[Function]}
                onClick={[Function]}
                style={
                  Object {
                    "display": "none",
                  }
                }
                tabIndex={-1}
                title="gallery"
                type="file"
              />
              <p>
                Drag 'n' drop some files here, or click to select files
              </p>
            </div>
          </section>
        </div>
      </div>
    </div>
  </fieldset>
</form>
`;

exports[`NumberField renders correctly 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="py-2"
  >
    <div
      className="relative"
    >
      <input
        className="bg-transparent border-gray-300 text-gray-600 peer h-10 border-0 border-b-2 placeholder-transparent focus:outline-none focus:ring-0 focus:border-red-900 w-36"
        id="ref"
        name="ref"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=" "
        step={1}
        type="number"
        value=""
      />
    </div>
  </div>
</form>
`;

exports[`SelectField renders correctly 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="py-2"
  >
    <div
      className="relative"
    >
      <div
        className="border-gray-300 border-0 border-b-2 relative w-36"
        data-testid="public-sizing-div"
      >
        <div
          className="peer css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-uuwi1j-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-319lph-ValueContainer"
            >
              <div
                className=" css-49e8sj-placeholder"
                id="react-select-2-placeholder"
              >
                Select
              </div>
              <input
                aria-autocomplete="list"
                aria-controls="react-select-2-listbox"
                aria-describedby="react-select-2-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-2-listbox"
                aria-readonly={true}
                className="css-mohuvp-dummyInput-DummyInput"
                disabled={false}
                id="react-select-2-input"
                inputMode="none"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                tabIndex={0}
                value=""
              />
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`SelectField renders correctly with multiple=true 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="py-2"
  >
    <div
      className="relative"
    >
      <div
        className="border-gray-300 border-0 border-b-2 relative w-100"
        data-testid="public-sizing-div"
      >
        <div
          className="peer css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-3-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-uuwi1j-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-319lph-ValueContainer"
            >
              <div
                className=" css-49e8sj-placeholder"
                id="react-select-3-placeholder"
              >
                Select
              </div>
              <input
                aria-autocomplete="list"
                aria-controls="react-select-3-listbox"
                aria-describedby="react-select-3-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-3-listbox"
                aria-readonly={true}
                className="css-mohuvp-dummyInput-DummyInput"
                disabled={false}
                id="react-select-3-input"
                inputMode="none"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                tabIndex={0}
                value=""
              />
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`SelectField sets basic theme correctly when prop is passed 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="py-2"
  >
    <div
      className="relative"
    >
      <div
        className="border-gray-300 border-0 border-b-2 relative w-36"
        data-testid="public-sizing-div"
      >
        <div
          className="peer css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-7-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-uuwi1j-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-319lph-ValueContainer"
            >
              <div
                className=" css-49e8sj-placeholder"
                id="react-select-7-placeholder"
              >
                Select
              </div>
              <input
                aria-autocomplete="list"
                aria-controls="react-select-7-listbox"
                aria-describedby="react-select-7-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-7-listbox"
                aria-readonly={true}
                className="css-mohuvp-dummyInput-DummyInput"
                disabled={false}
                id="react-select-7-input"
                inputMode="none"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                tabIndex={0}
                value=""
              />
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`SelectField sets basic theme correctly when prop is passed for a multi-select 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="py-2"
  >
    <div
      className="relative"
    >
      <div
        className="border-gray-300 border-0 border-b-2 relative w-100"
        data-testid="public-sizing-div"
      >
        <div
          className="peer css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-9-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-uuwi1j-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-319lph-ValueContainer"
            >
              <div
                className=" css-49e8sj-placeholder"
                id="react-select-9-placeholder"
              >
                Select
              </div>
              <input
                aria-autocomplete="list"
                aria-controls="react-select-9-listbox"
                aria-describedby="react-select-9-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-9-listbox"
                aria-readonly={true}
                className="css-mohuvp-dummyInput-DummyInput"
                disabled={false}
                id="react-select-9-input"
                inputMode="none"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                tabIndex={0}
                value=""
              />
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`SelectField sets full theme correctly when prop is passed 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="py-2"
  >
    <div
      className="relative"
    >
      <div
        className="border-gray-300 border-0 border-b-2 relative w-36"
        data-testid="public-sizing-div"
      >
        <div
          className="peer css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-8-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-uuwi1j-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-319lph-ValueContainer"
            >
              <div
                className=" css-49e8sj-placeholder"
                id="react-select-8-placeholder"
              >
                Select
              </div>
              <input
                aria-autocomplete="list"
                aria-controls="react-select-8-listbox"
                aria-describedby="react-select-8-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-8-listbox"
                aria-readonly={true}
                className="css-mohuvp-dummyInput-DummyInput"
                disabled={false}
                id="react-select-8-input"
                inputMode="none"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                tabIndex={0}
                value=""
              />
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`SelectField sets full theme correctly when prop is passed for a multi-select 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="py-2"
  >
    <div
      className="relative"
    >
      <div
        className="border-gray-300 border-0 border-b-2 relative w-100"
        data-testid="public-sizing-div"
      >
        <div
          className="peer css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-10-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-uuwi1j-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-319lph-ValueContainer"
            >
              <div
                className=" css-49e8sj-placeholder"
                id="react-select-10-placeholder"
              >
                Select
              </div>
              <input
                aria-autocomplete="list"
                aria-controls="react-select-10-listbox"
                aria-describedby="react-select-10-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-10-listbox"
                aria-readonly={true}
                className="css-mohuvp-dummyInput-DummyInput"
                disabled={false}
                id="react-select-10-input"
                inputMode="none"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                tabIndex={0}
                value=""
              />
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`SubmitButton renders correctly 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <button
    className="disabled:opacity-70 bg-lime-200 text-lime-600 hover:opacity-90 w-min my-4 px-14 py-2 text-center text-xl font-bold rounded shadow-xl"
    type="submit"
  >
    create new record
  </button>
</form>
`;

exports[`TextAreaField renders correctly 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="py-2"
  >
    <div
      className="relative"
    >
      <textarea
        className="bg-transparent border-gray-300 text-gray-600 peer h-10 border-0 border-b-2 placeholder-transparent focus:outline-none focus:ring-0 focus:border-red-900 w-100 focus:h-32"
        id="title"
        name="title"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=" "
        value=""
      />
    </div>
  </div>
</form>
`;

exports[`TextField renders correctly 1`] = `
<form
  action="#"
  name="test-schema"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="py-2"
  >
    <div
      className="relative"
    >
      <input
        className="bg-transparent border-gray-300 text-gray-600 peer h-10 border-0 border-b-2 placeholder-transparent focus:outline-none focus:ring-0 focus:border-red-900 w-full"
        id="title"
        name="title"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=" "
        type="text"
        value=""
      />
    </div>
  </div>
</form>
`;
